src/basic/env-util.c

```diff
@@ -884,9 +884,13 @@ int replace_env_argv(
                 char ***ret_bad_variables) {
 
         _cleanup_strv_free_ char **n = NULL, **unset_variables = NULL, **bad_variables = NULL;
-        size_t k = 0, l = 0;
+        size_t k = 0, l;
         int r;
 
+        assert(!strv_isempty(argv));
+        assert(env);
```

Comment:
hmm, an env block should be allowed to be empty, and we generally allow empty strv to be encoded as simple NULL...